crates/router/src/configs/settings.rs

```diff
@@ -553,7 +553,7 @@ pub struct DelayedSessionConfig {
 
 #[derive(Debug, Deserialize, Clone, Default)]
 pub struct ConnectorRequestReferenceIdConfig {
-    pub merchant_ids_send_payment_id_as_connector_request_id: HashSet<String>,
+    pub merchant_ids_send_payment_id_as_connector_request_id: Vec<String>,
```

Comment:
why this?
Does config crate not support reading hashset from comma separated envs?